services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/BandwidthSpec.groovy

```diff
@@ -68,6 +69,51 @@ class BandwidthSpec extends BaseSpecification {
         flow?.id && northboundService.deleteFlow(flow.id)
     }
 
+    def "Verify longer path is chosen in case of not enough available bandwidth on shorter path"() {
+        given: "Two active switches and two possible flow paths at least"
+        def switches = topology.getActiveSwitches()
+        List<List<PathNode>> possibleFlowPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            possibleFlowPaths = topologyEngineService.getPaths(src.dpId, dst.dpId)*.path.sort { it.size() }
+            possibleFlowPaths.size() > 1
+        }
+        assert srcSwitch && dstSwitch
+
+        // Make the first path more preferable than others.
+        possibleFlowPaths[1..-1].each { pathHelper.makePathMorePreferable(possibleFlowPaths.first(), it) }
+
+        // Get min available bandwidth on the preferable path.
+        def involvedBandwidths = []
+        def allLinks = northboundService.getAllLinks()
+        pathHelper.getInvolvedIsls(possibleFlowPaths.first()).each {
+            involvedBandwidths.add(islUtils.getIslInfo(allLinks, it).availableBandwidth)
+        }
+        def minAvailableBandwidth = involvedBandwidths.min()
+
+        // Create a flow to reduce available bandwidth on links of the preferable path.
```

Comment:
If you open your spock report for this test, will it be clear what is going on in this test?

Reply:
Let's improve it in another patch.